google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreOptions.java

```diff
@@ -119,6 +159,32 @@ private DatastoreOptions(Builder builder) {
     super(DatastoreFactory.class, DatastoreRpcFactory.class, builder, new DatastoreDefaults());
     namespace = MoreObjects.firstNonNull(builder.namespace, defaultNamespace());
     databaseId = MoreObjects.firstNonNull(builder.databaseId, DEFAULT_DATABASE_ID);
+
+    // todo see if we can update this
+    if (getTransportOptions() instanceof HttpTransportOptions
+        && (builder.channelProvider != null || builder.credentialsProvider != null)) {
+      throw new IllegalArgumentException(
+          "Only gRPC transport allows setting of channel provider or credentials provider");
+    } else {
+      this.channelProvider =
```

Comment:
Do we still want to instantiate this to be GrpcTransportProvider if `(getTransportOptions() instanceof HttpTransportOptions) && (builder.channelProvider == null && builder.credentialsProvider == null) == true`?

Reply:
great catch, thank you! I updated the logic and added a test to confirm it's working as expected